spec/shared/context/leadprovider_profiles_and_courses.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+RSpec.shared_context "leadprovider profiles and courses" do
+  let(:cpd_lead_provider) { create(:cpd_lead_provider, :with_lead_provider) }
+  let(:another_lead_provider) { create(:cpd_lead_provider, name: "Unknown") }
+  let(:ect_profile) { create(:early_career_teacher_profile) }
+  let(:mentor_profile) { create(:mentor_profile) }
+  let(:induction_coordinator_profile) { create(:induction_coordinator_profile) }
+  let(:npq_lead_provider) { create(:npq_lead_provider, cpd_lead_provider: cpd_lead_provider) }
+  let(:npq_course) { create(:npq_course, identifier: "npq-leading-teaching") }
+  let!(:npq_profile) do
+    create(:npq_validation_data,
+           npq_lead_provider: npq_lead_provider,
+           npq_course: npq_course)
+  end
+
+  let(:delivery_partner) { create(:delivery_partner) }
+  let!(:school_cohort) { create(:school_cohort, school: ect_profile.school, cohort: ect_profile.cohort) }
+  let!(:partnership) do
+    create(:partnership,
+           school: ect_profile.school,
+           lead_provider: cpd_lead_provider.lead_provider,
+           cohort: ect_profile.cohort,
+           delivery_partner: delivery_partner)
+  end
+
+  def generate_raw_event(params)
+    params.except(:raw_event, :cpd_lead_provider).to_json
+  end
```

Comment:
That probably does not belong in this file - cause this file does not know about params

Reply:
The other shared context likely wants it more :)